### ISSUE TITLE
Fix DDP Example for CIFAR10 by Using Epochs Only

### ISSUE THREAD

**Issue by ImahnShekhzadeh** [atong01/conditional-flow-matching#147]
Hi!

As recently discussed in #145 and #144 with @Xiaoming-Zhao (and I as had already mentioned in https://github.com/atong01/conditional-flow-matching/pull/116#discussion_r1695722539), I/we believe that it would be best to use epochs instead of steps in the distributed mode, i.e. in the training loop of `train_cifar10_ddp.py`. Is there a reason not to (except consistency to the other scripts)? :upside_down_face:

I would be happy to make the changes, test them extensively and then open a PR.

Cheers,
Imahn

**Comment by atong01** [comment]
Yes I agree with this change in principle. Happy to take a PR in this direction. Thanks for all your work on this example!

**Comment by kilianFatras** [comment]
Of course. I might do it if I have time but it is rather uncertain for the end of the year.
